### PR TITLE
[S3UTILS-130] ✨ Implement GHAS

### PR DESCRIPTION
**Pull request by gaspardmoindrot**

### CODE REVIEW

File: .github/workflows/dependency-review.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+jobs:
+  dependency-review:
+    runs-on: ubuntu-latest
+    steps:
+      - name: 'Checkout Repository'
+        uses: actions/checkout@v3
+
+      - name: 'Dependency Review'
+        uses: actions/dependency-review-action@v3
```
Comment (gaspardmoindrot) [inline .github/workflows/dependency-review.yaml]:
```suggestion
        uses: actions/dependency-review-action@v3

```

File: .github/workflows/security.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+       REGISTRY_PASSWORD: ${{ secrets.REGISTRY_PASSWORD }}
+
+   trivy:
+     needs: build
+     name: Trivy Vulnerability Scan for Code
+     uses: scality/workflows/.github/workflows/trivy.yaml@v1
+     with:
+       name: ${{ github.event.repository.name }}
+       namespace: s3utils-dev
+       registry: registry.scality.com
```
Comment (gaspardmoindrot) [inline .github/workflows/security.yaml]:
```suggestion
       registry: registry.scality.com

```

File: .github/workflows/security.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+ jobs:
+   build:
+     permissions:
+       contents: read
+       packages: write
+     uses: ./.github/workflows/build.yml
+     secrets:
+       REGISTRY_LOGIN: ${{ secrets.REGISTRY_LOGIN }}
+       REGISTRY_PASSWORD: ${{ secrets.REGISTRY_PASSWORD }}
```
Comment (tcarmet) [inline .github/workflows/security.yaml]: The secrets are badly defined in the workflow, use this method instead:
```suggestion
     secrets: inherit
```

**Review by tcarmet — CHANGES_REQUESTED** [review]

**Review by tcarmet — APPROVED** [review]

**Comment by gaspardmoindrot** [comment]
`ping`

**Comment by gaspardmoindrot** [comment]
`ping`

**Review by tmacro — APPROVED** [review]

**Comment by gaspardmoindrot** [comment]
`ping`

**Pull request merged**